modules/@apostrophecms/schema/ui/apos/lib/detectChange.js

```diff
@@ -44,7 +44,7 @@ export function detectFieldChange(field, v1, v2) {
   if (isEqual(v1, v2)) {
     return false;
   } else if (!v1 && !v2) {
-    return false;
+    return v1 !== v2;
```

Comment:
Please explicitly check for that one difference then and "else if" use the old test. This could have many knock-on effects and we want to minimize it.

Reply:
Do you mean that?

```js
  } else if (!v1 && !v2) {
    return v1 !== v2;
  } else if (!v1 && !v2) {
    return false;
  }
```

because it will never reach the 2nd `else if` statement if we enter the first one

Reply:
It's fixing the case when we have `null` and `false` (cf 1st [comment](https://github.com/apostrophecms/apostrophe/pull/4118/files#r1157339181) in this thread), so both of these will be falsy, but they are different.

Reply:
I am concerned that there may be other pairs of falsy values that should not be considered different, but let's see how it goes this way, it's more correct ultimately.